### PR TITLE
Allowing command module authors to inject formatter class.

### PR DESCRIPTION
**Pull request by scobb**

### CODE REVIEW

**Comment by tjprescott** [comment]
What is the intent behind exposing this functionality?

**Comment by scobb** [comment]
Our specific use case is that we need to allow users to specify argument values that begin with a hyphen. The only way to accomplish this is to use the equal version of argument specification: 
--in=--some-user-value

We want the help messages to guide users to this--can get some of the functionality using metavar, but to force the equal to appear, we have to override the help parser. 

Current help message without overriding HelpFormatter (notice no '=' after the first '--in')
usage: az ml service create batch [-h] [--output {json,tsv,table,jsonc}]
                                  [--in <input_name>:[<default_value>]
                                  [--in=...]]

Our desired help message is (notice the '=' after first '--in'):
usage: az ml service create batch [-h] [--output {json,tsv,table,jsonc}]
                                  [--in=<input_name>:[<default_value>]
                                  [--in=...]]

File: src/azure-cli-core/azure/cli/core/parser.py [hunk 1]
```diff
@@ -68,10 +68,17 @@ def load_command_table(self, command_table):
             # To work around http://bugs.python.org/issue9253, we artificially add any new
             # parsers we add to the "choices" section of the subparser.
             subparser.choices[command_verb] = command_verb
+
+            # inject command_module designer's help formatter -- default is HelpFormatter
+            fc = metadata.formatter_class if metadata.formatter_class else \
```
Comment (troydai) [inline src/azure-cli-core/azure/cli/core/parser.py]:
```
fc = metadata.formatter_class or argparse.HelpFormatter
```

Comment (scobb) [inline src/azure-cli-core/azure/cli/core/parser.py]: Updated. Thanks for the suggestion.

**Comment by troydai** [comment]
The change itself looks fine to me. @johanste you opinion?

**Comment by johanste** [comment]
The changes look good - my only concern is that we are bypassing the formatter for some of the general help output. But as long as the help provided when providing --help to the command looks good, I'm fine with this change!

**Pull request merged**